src/npc_attack.cpp

```diff
@@ -486,23 +600,28 @@ npc_attack_rating npc_attack_throw::evaluate_tripoint(
     if( critter ) {
         att = source.attitude_to( *critter );
     }
-    const bool friendly_fire = att == Creature::Attitude::FRIENDLY &&
-                               !source.rules.has_flag( ally_rule::avoid_friendly_fire );
-    int attitude_mult = npc_attack_constants::attitude_multiplier.at( att );
-    if( friendly_fire ) {
-        // hitting a neutral creature isn't exactly desired, but it's a lot less than a friendly.
-        // if friendly fire is on, we don't care too much, though if an available hit doesn't damage them it would be better.
-        attitude_mult = npc_attack_constants::attitude_multiplier.at( Creature::Attitude::NEUTRAL );
+
+    if( att != Creature::Attitude::HOSTILE ) {
+        // No point in throwing stuff at neutral/friendly targets
+        return npc_attack_rating( cata::nullopt, location );
+    }
+
+    if( source.rules.has_flag( ally_rule::avoid_friendly_fire ) &&
+        !source.wont_hit_friend( location, thrown_item, false ) ) {
+        // Avoid friendy fire
+        return npc_attack_rating( cata::nullopt, location );
     }
 
     const float throw_mult = throw_cost( source, single_item ) * source.speed_rating() / 100.0f;
     const int damage = source.thrown_item_total_damage_raw( single_item );
     float dps = damage / throw_mult;
     const int distance_to_me = rl_dist( location, source.pos() );
-    const int distance_penalty = std::max( distance_to_me - 1,
-                                           ( source.closest_enemy_to_friendly_distance() - 1 ) * 2 );
+    const int distance_penalty = std::max( std::min( distance_to_me,
+                                           source.closest_enemy_to_friendly_distance() ) - 3, 0 );
+    const float suitable_item_mult = thrown_item.has_flag( flag_NPC_THROWN ) ? 0.2f : -0.15f;
```

Comment:
If this is a melee/true ranged weapon without flag_NPC_THROW, perhaps base this penalty on the weapon dps?

Reply:
At first I tried to add a check to see if the weapon had higher dps when wielded than thrown but it didn't work out as expected. I eventually reduced the multiplier for items not flagged and it made the NPCs not throw weapons as often. I think cudgels are still thrown in some circunstamces but that's why the math still needs a lot of work.

It's mainly the logic when executing actions that I fixed but the math is just placeholder and random values that simply worked for my tests.